airflow/providers/smtp/notifications/smtp.py

```diff
@@ -0,0 +1,106 @@
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from typing import Any, Iterable
+
+from airflow.compat.functools import cached_property
+from airflow.exceptions import AirflowOptionalProviderFeatureException
+
+try:
+    from airflow.notifications.basenotifier import BaseNotifier
+except ImportError:
+    raise AirflowOptionalProviderFeatureException(
+        "Failed to import BaseNotifier. This feature is only available in Airflow versions >= 2.6.0"
+    )
+
+from airflow.providers.smtp.hooks.smtp import SmtpHook
+
+
+class SmtpNotifier(BaseNotifier):
+    """
+    SMTP Notifier
+
+    :param smtp_conn_id: The :ref:`smtp connection id <howto/connection:smtp>`
+        that contains the information used to authenticate the client.
+    """
+
+    template_fields = (
+        "mail_from",
+        "to",
+        "subject",
+        "html_content",
+        "files",
+        "cc",
+        "bcc",
+        "mime_subtype",
+        "mime_charset",
+        "custom_headers",
+    )
+
+    def __init__(
+        self,
+        mail_from: str | None,
+        to: str | Iterable[str],
+        subject: str,
+        html_content: str,
+        files: list[str] | None = None,
+        cc: str | Iterable[str] | None = None,
+        bcc: str | Iterable[str] | None = None,
+        mime_subtype: str = "mixed",
+        mime_charset: str = "utf-8",
+        custom_headers: dict[str, Any] | None = None,
+        smtp_conn_id: str = SmtpHook.default_conn_name,
+    ):
+        super().__init__()
+        self.smtp_conn_id = smtp_conn_id
+        self.mail_from = mail_from
+        self.to = to
+        self.subject = subject
+        self.html_content = html_content
+        self.files = files
+        self.cc = cc
+        self.bcc = bcc
+        self.mime_subtype = mime_subtype
+        self.mime_charset = mime_charset
+        self.custom_headers = custom_headers
+
+    @cached_property
+    def hook(self) -> SmtpHook:
+        """Smtp Events Hook"""
+        return SmtpHook(smtp_conn_id=self.smtp_conn_id)
+
+    def notify(self, context):
+        """Send a email via smtp server"""
+        with self.hook as smtp:
+            smtp.send_email_smtp(
+                smtp_conn_id=self.smtp_conn_id,
+                mail_from=self.mail_from,
```

Comment:
```suggestion
                mail_from=self.mail_from,
```

This should be `from_email`. Make sure you test on a connection without a [from_email set](https://github.com/apache/airflow/blob/c5597d1fabe5d8f3a170885f6640344d93bf64bf/airflow/providers/smtp/hooks/smtp.py#L327).

Reply:
I have no idea why `send_email_smtp` accepts extra kwargs. It just hid this issue :(

Reply:
Ya, that hid the issue, made the changes suggested, and also made sure the `from_email` is picked when passed in `send_smtp_notification`.